src/workflow/tools.ts

```
@@ -0,0 +1,111 @@
+  const request = (summary: ApprovalSummary, sessionID: string, context: ToolContext) => new Promise<Readonly<{ decision: PolicyDecision; token?: string }>>(async (resolve) => {
+    const requestID = randomUUID()
+    const timer = setTimeout(() => { pending.delete(requestID); resolve({ decision: { allowed: false, layer: "approval", reason: "Approval reply timed out" } }) }, options.timeoutMs ?? 60_000)
+    pending.set(requestID, { resolve: (decision) => { clearTimeout(timer); pending.delete(requestID); const token = decision.allowed ? `${requestID}.${hash({ requestID, summary, sessionID })}` : undefined; if (token) issued.set(token, { workflowHash: summary.workflowHash, policyHash: summary.policyHash, sessionID, expiresAt: Date.now() + 60_000 }); resolve({ decision, token }) } })
+    const onAbort = () => { if (pending.has(requestID)) pending.get(requestID)!.resolve({ allowed: false, layer: "approval", reason: "Approval request aborted" }) }
+    if (context.abort.aborted) onAbort(); else context.abort.addEventListener("abort", onAbort, { once: true })
+    try {
+      const redacted = { ...summary, commands: summary.commands.map((c) => ({ executable: c.executable, argv: c.argv, cwd: c.cwd })) }
+      await context.ask({ permission: "workflow.launch", patterns: [summary.workflowHash], always: [summary.workflowHash], metadata: { requestID, workflowHash: summary.workflowHash, policyHash: summary.policyHash, sessionID, summary: redacted } })
+    }
+    catch { clearTimeout(timer); pending.delete(requestID); resolve({ decision: { allowed: false, layer: "approval", reason: "Approval request failed" } }) }
+  })
```

Biome (2.5.7)

[error] Promise executor functions should not be `async`.

(lint/suspicious/noAsyncPromiseExecutor)